src/main/java/strategy/handlers/SupplyHandler.java

```diff
@@ -0,0 +1,12 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class SupplyHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.fruitStorage.merge(transaction.getFruit(), transaction.getQuantity(), Integer::sum);
+    }

```

Comment:
According to the encapsulation principle, consider making `SupplyHandler` class itself package-private if it's not intended to be used outside of its package, and ensure that all members are using the least permissive access level necessary.